crates/indexer/src/util.rs

```diff
@@ -12,33 +12,63 @@ use {
 
 /// Borrow a `solana-sdk` account as a `solana-program` account info struct.
 #[cfg(feature = "solana-program")]
-pub fn account_as_info<'a>(
-    key: &'a Pubkey,
+pub async fn account_as_info<T: Send + 'static>(
+    key: Pubkey,
     is_signer: bool,
     is_writable: bool,
-    acct: &'a mut Account,
-) -> AccountInfo<'a> {
-    AccountInfo::new(
-        key,
-        is_signer,
-        is_writable,
-        &mut acct.lamports,
-        &mut *acct.data,
-        &acct.owner,
-        acct.executable,
-        acct.rent_epoch,
-    )
+    mut acct: Account,
+    f: impl Send + FnOnce(AccountInfo<'_>) -> T + 'static,
+) -> Result<T> {
+    // NOTE: this is here because metaplex_auction makes the brilliant choice to
+    //       only provide parsing via AccountInfo, which further makes the
+    //       brilliant choice to store lamports as Rc<RefCell<&mut u64>>, which
+    //       does not implement Send and therefore cannot ever be used in
+    //       multithreaded code.  Thus the only way to access it is to spawn a
+    //       dedicated thread in Tokio's thread pool to run all code that needs
+    //       to use the account info and prevent the need for a Send impl.
```

Comment:
```suggestion
    // NOTE: this is here because metaplex_auction
    //       only provide parsing via AccountInfo, and also stores lamports as Rc<RefCell<&mut u64>>, which
    //       does not implement Send and therefore cannot ever be used in
    //       multithreaded code.  Thus the only way to access it is to spawn a
    //       dedicated thread in Tokio's thread pool to run all code that needs
    //       to use the account info and prevent the need for a Send impl.
```